### PR TITLE
Error out if target database is missing any objects

### PR DESCRIPTION
**Pull request by lavjain**
during incremental restore unless --on-error-continue

### CODE REVIEW

File: restore/restore.go [hunk 1]
```diff
@@ -224,24 +228,30 @@ func restorePredata(metadataFilename string) {
 				if utils.RelationIsExcludedByUser(inRelationsUserInput, exRelationsUserInput, table) {
 					tablesExcludedByUserInput = append(tablesExcludedByUserInput, table)
 				} else {
-					if !utils.Exists(schemasToCreate, schemaName) {
+					_, ok := existingSchemasMap[schemaName]
```
Comment (roicos) [inline restore/restore.go:231]:
```suggestion
					_, schemaNotExists := !existingSchemasMap[schemaName]
					prefilteredToCreate := !utils.Exists(schemasToCreate, schemaName)
					if schemaNotExists && !prefilteredToCreate {
```

File: restore/restore.go [hunk 2]
```diff
@@ -296,8 +306,8 @@ func restoreData() {
 	restorePlan := backupConfig.RestorePlan
 	restorePlanEntries := make([]history.RestorePlanEntry, 0)
 	if MustGetFlagBool(options.INCREMENTAL) {
-		restorePlanEntries = append(restorePlanEntries, restorePlan[len(backupConfig.RestorePlan)-1])
-
+		restorePlanEntries = append(restorePlanEntries,
```
Comment (shivzone) [inline restore/restore.go:309]: This change seems unnecessary

File: restore/restore.go [hunk 3]
```diff
@@ -224,24 +228,30 @@ func restorePredata(metadataFilename string) {
 				if utils.RelationIsExcludedByUser(inRelationsUserInput, exRelationsUserInput, table) {
 					tablesExcludedByUserInput = append(tablesExcludedByUserInput, table)
 				} else {
-					if !utils.Exists(schemasToCreate, schemaName) {
+					_, ok := existingSchemasMap[schemaName]
+					if !ok && !utils.Exists(schemasToCreate, schemaName) {
```
Comment (shivzone) [inline restore/restore.go:232]: Can make the whole thing a one liner to reduce clutter `if _, val := existingSchemasMap[schemaName]; !val && !utils.Exists(schemasToCreate, schemaName)`

File: restore/restore.go [hunk 4]
```diff
@@ -201,6 +201,10 @@ func restorePredata(metadataFilename string) {
 		existingTableFQNs, err := GetExistingTableFQNs()
 		gplog.FatalOnError(err)
 
+		existingSchemasMap := make(map[string]Empty)
```
[the remark this answers is not in the document]
Comment (shivzone) [reply restore/restore.go:204]: utils.exists() is a linear vs constant time map lookup

[the remark this answers is not in the document]
Comment (shivzone) [reply restore/restore.go:204]: In some scenarios we use it for a rather small input like `includeSchemas` which are rather small and using a map is an overkill.

**Pull request merged**